### PR TITLE
Remove CAPI release-1.3 branch jobs and dashboards

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
This patch removes CAPI presubmit/periodics/periodic-upgrade jobs + dashboards for release-1.3 branch.
/hold until release-1.6 is out and we are ready to drop the tests

### CODE REVIEW

**Comment by adilGhaffarDev** [comment]
cc @furkatgofurov7 @nawazkh @killianmuldoon

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

I think we should be happy to drop these once `v1.6.0-RC.0` is released (planned for Tuesday, November 14th).

**Review by killianmuldoon — COMMENTED** [review]
Note: We should consider moving `release-1.4` tests to the eks cluster in order to keep some coverage there. 

We can either do that at the same time as this change or after v1.4.X is cut at the end of the month. I'll open an issue to track this.

**Comment by adilGhaffarDev** [comment]
> Note: We should consider moving `release-1.4` tests to the eks cluster in order to keep some coverage there.
> 
> We can either do that at the same time as this change or after v1.4.X is cut at the end of the month. I'll open an issue to track this.

Agree, I was also thinking about it. I will create PR for that too once this test is dropped.

File: config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml [hunk 1]
```diff
@@ -5,7 +5,6 @@ dashboard_groups:
     - sig-cluster-lifecycle-kubeadm
     - sig-cluster-lifecycle-cluster-addons
     - sig-cluster-lifecycle-cluster-api
-    - sig-cluster-lifecycle-cluster-api-1.3
```
Comment (killianmuldoon) [inline config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml:8]: Also needs to be remove on L37

Comment (adilGhaffarDev) [reply config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml:8]: sorry I missed that, fixed it now.

**Comment by furkatgofurov7** [comment]
> Note: We should consider moving `release-1.4` tests to the eks cluster in order to keep some coverage there.
> 
> We can either do that at the same time as this change or after v1.4.X is cut at the end of the month. I'll open an issue to track this.
> 
> Edit: Mentioned this here: [kubernetes-sigs/cluster-api#9609](https://github.com/kubernetes-sigs/cluster-api/issues/9609)

Fully agree with ^, while changes look good to me.

/lgtm

**Comment by furkatgofurov7** [comment]
Echoing https://github.com/kubernetes-sigs/cluster-api/issues/9609#issuecomment-1810084379, let's keep them for now until we have some more jobs running in EKS

**Pull request merged**